### DISCUSSION TITLE
Replace variables should include id in plugin.json

### DISCUSSION THREAD

**Discussion by JorundMartinsen** [grafana/grafana#39417] [Plugins]
Currently, 'version' and 'updated' are set during the webpack run.

Version is handled with a search and replace as below. 
This should also include the id of the plugin, as an opt-in. 
This should not break existing plugins, and would make referencing simpler
```
new ReplaceInFileWebpackPlugin([
      {
        dir: 'dist',
        files: ['plugin.json', 'README.md'],
        rules: [
// ADD THIS
//          {
//            search: '%NAME%',
//            replace: packageJson.name,
//          },
          {
            search: '%VERSION%',
            replace: packageJson.version,
          },
          {
            search: '%TODAY%',
            replace: new Date().toISOString().substring(0, 10),
          },
        ],
      },
    ]),
```

This would trigger a change in the schema for plugin.json, documentation and possibly the plugin template

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
